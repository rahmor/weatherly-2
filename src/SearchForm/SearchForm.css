.SearchForm {
  padding-top: 1rem;
}

.SearchForm > * {
  color: white;
  background-color: rgba(135, 207, 235, 0);
}

.SearchForm__LocationArrow {
  font-size: x-small;
}

.SearchForm__Search {
  font-size: small;
}

.SearchForm > h6 {
  padding: 0.5rem 0 0.5rem 0;
  margin: 0;
}

.SearchForm input {
  background-color: rgba(135, 207, 235, 0);
  border: 0;
  color: white;
}

.SearchForm input::placeholder {
  color: white;
}

.SearchForm button {
  background-color: rgba(135, 207, 235, 0);
  border: 0;
  color: white;
}
