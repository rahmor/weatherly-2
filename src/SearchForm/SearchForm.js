import React, { useState, useEffect } from 'react';
import moment from 'moment';
import { Field, reduxForm } from 'redux-form';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSearch, faLocationArrow } from '@fortawesome/free-solid-svg-icons';
import './SearchForm.css';

let SearchForm = (props) => {
  /*
  Get random city on loading and put in useState
  after lazy loading of info.  Info return will
  call action creator with returned info and will
  put all info into Redux store.

   *
   */
  const [CITIES] = useState({
    'New York': [40.7127281, -74.0060152],
    Atlanta: [33.7490987, -84.3901849],
    'Los Angeles': [34.0536909, -118.2427666],
    Miami: [25.7741728, -80.19362],
    'San Francisco': [37.7790262, -122.4199061],
    Chicago: [41.8755616, -87.6244212],
    'Washington, DC': [38.8949855, -77.0365708],
    Boston: [42.3602534, -71.0582912],
  });

  let [inputting, setInputting] = useState(false);
  console.log(props.city);
  return (
    <header className='SearchForm'>
      <form
        id='search'
        name='search'
        initialValues={{ city: props.city }}
        onSubmit={(e) => {
          e.preventDefault();
          setInputting((prevState) => !prevState);
        }}
      >
        <label className='App__Visually__Hidden' htmlFor='city'></label>
        <FontAwesomeIcon
          className='SearchForm__LocationArrow'
          icon={faLocationArrow}
        >
          Search by City
        </FontAwesomeIcon>
        {inputting ? (
          <Field
            id='search'
            name='city'
            component='input'
            type='text'
            placeholder={props.city}
          />
        ) : (
          <p
            onClick={() => {
              setInputting((prevState) => !prevState);
            }}
          >
            {props.city}
          </p>
        )}

        <button type='submit'>
          <FontAwesomeIcon
            className='SearchForm__Search'
            icon={faSearch}
          ></FontAwesomeIcon>
        </button>
      </form>
      <h6>{moment().format('MMMM Do YYYY, h:mm a')}</h6>
    </header>
  );
};

SearchForm = reduxForm({
  form: 'form',
})(SearchForm);

export default SearchForm;
