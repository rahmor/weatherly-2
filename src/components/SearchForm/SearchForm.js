import React, { useState } from 'react';
import moment from 'moment';
import { connect } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSearch, faLocationArrow } from '@fortawesome/free-solid-svg-icons';
import { Field, reduxForm, formValueSelector } from 'redux-form';
import './SearchForm.css';

export let SearchFormFields = (props) => {
  let [inputting, setInputting] = useState(false);

  return (
    <form
      id='search'
      name='search'
      onSubmit={(e) => {
        e.preventDefault();
        setInputting((prevState) => !prevState);
        props.fetchCoordinates(props.citySearch);
      }}
    >
      <label className='App__Visually__Hidden' htmlFor='city'></label>
      <FontAwesomeIcon
        className='SearchForm__LocationArrow'
        icon={faLocationArrow}
      >
        Search by City
      </FontAwesomeIcon>
      {inputting ? (
        <Field
          name='citySearch'
          component='input'
          type='text'
          placeholder={props.city}
        ></Field>
      ) : (
        <span
          onClick={() => {
            setInputting((prevState) => !prevState);
          }}
        >
          {props.city}
        </span>
      )}

      <button type='submit'>
        <FontAwesomeIcon
          className='SearchForm__Search'
          icon={faSearch}
        ></FontAwesomeIcon>
      </button>
    </form>
  );
};

SearchFormFields = reduxForm({
  form: 'search',
})(SearchFormFields);

const SearchForm = (props) => {
  return (
    <header className='SearchForm'>
      <SearchFormFields {...props} />
      <p className='SearchForm__Date'>
        {moment().format('MMMM Do YYYY, h:mm a')}
      </p>
    </header>
  );
};

const selector = formValueSelector('search');

SearchFormFields = connect((state) => {
  const citySearch = selector(state, 'citySearch');
  return {
    citySearch,
  };
})(SearchFormFields);

export default SearchForm;
