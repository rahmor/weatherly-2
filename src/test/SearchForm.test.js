import SearchForm from '../components/SearchForm/SearchForm';
import React from 'react';
import moment from 'moment';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { after } from 'lodash';
jest.mock('moment', () => {
  return () => jest.requireActual('moment')('2020-01-01T00:00:00.000Z');
});

describe('SearchForm component', () => {
  const props = { fetchCoordinates: jest.fn(), city: 'New York' };
  const wrapper = shallow(<SearchForm {...props} />);

  afterAll(() => {
    wrapper.unmount();
    wrapper.debug();
  });

  it('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe('when span clicked', () => {
    it('should match snapshot and show input element', () => {
      wrapper.find('span').simulate('click');
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  it('should call renderProps', () => {
    wrapper
      .find('form')
      .at(0)
      .simulate('submit', { preventDefault: jest.fn() });
    expect(props.fetchCoordinates).toHaveBeenCalled();
  });
});
