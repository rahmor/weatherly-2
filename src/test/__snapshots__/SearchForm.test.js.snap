// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component should match snapshot 1`] = `
<header
  className="SearchForm"
>
  <Connect(ReduxForm) />
  <p
    className="SearchForm__Date"
  >
    December 31st 2019, 7:00 pm
  </p>
</header>
`;
